symengine/tests/basic/test_sets.cpp

```diff
@@ -494,8 +507,8 @@ TEST_CASE("Complement : Basic", "[basic]")
 
     CHECK_THROWS_AS(r2->set_intersection(finiteset({symbol("x")})),
                     std::runtime_error);
-    r1 = r2->set_intersection(finiteset({zero, integer(2)}));
-    REQUIRE(eq(*r1, *finiteset({zero, integer(2)})));
+    CHECK_THROWS_AS(r2->set_intersection(finiteset({zero, integer(2)})),
+                    std::runtime_error);
```

Comment:
similar argument as ^above here.
`r2` contains `(-oo, oo) \ {y}`. so we can't simplify this unless we have info of `y`. So, this should return an object of type `Intersection` class which is currently not implemented. So, `runtime_error`.